app/src/main/java/com/zeapo/pwdstore/git/BreakOutOfDetached.kt

```diff
@@ -0,0 +1,73 @@
+package com.zeapo.pwdstore.git
+
+import android.app.Activity
+import android.app.AlertDialog
+import com.zeapo.pwdstore.R
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.GitCommand
+import org.eclipse.jgit.api.PushCommand
+import org.eclipse.jgit.api.RebaseCommand
+import java.io.File
+
+class BreakOutOfDetached(fileDir: File, callingActivity: Activity) : GitOperation(fileDir, callingActivity) {
+    private lateinit var commands: List<GitCommand<out Any>>
+
+    /**
+     * Sets the command
+     *
+     * @return the current object
+     */
+    fun setCommands(): BreakOutOfDetached {
+        val git = Git(repository)
+        val branchName = "conflicting-master-${System.currentTimeMillis()}"
+
+        this.commands = listOf(
+                // abort the rebase
+                git.rebase().setOperation(RebaseCommand.Operation.ABORT),
+                // git checkout -b conflict-branch
+                git.checkout().setCreateBranch(true).setName(branchName),
+                // push the changes
+                git.push().setRemote("origin"),
+                // switch back to master
+                git.checkout().setName("master")
+        )
+        return this
+    }
+
+    override fun execute() {
+        val git = Git(repository)
+        if (!git.repository.repositoryState.isRebasing) {
+            AlertDialog.Builder(callingActivity)
```

Comment:
Reminds me to migrate all of these MaterialAlertDialog. Will follow up once this lands.